iree/samples/static_library/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+################################################################################
+# Autogenerated by build_tools/bazel_to_cmake/bazel_to_cmake.py from           #
+# iree/samples/simple_embedding/BUILD                                          #
+#                                                                              #
+# Use iree_cmake_extra_content from iree/build_defs.oss.bzl to add arbitrary   #
+# CMake-only content.                                                          #
+#                                                                              #
+# To disable autogeneration for this file entirely, delete this header.        #
+################################################################################
+
+iree_add_all_subdirs()
+
+iree_c_embed_data(
```

Comment:
we should be able to generate this in the build as we do other artifacts (iree_bytecode_module) - having this all tested on CI is critical to avoiding rot

Reply:
Would be nice if we could fit the `-iree-llvm-static-library-output-path=iree/samples/static_library/static_library_module.o` flag and `${CMAKE_CURRENT_SOURCE_DIR}/static_library_module.o` (generated) dep into our current config with `iree_bytecode_module`.

Otherwise, this could slot into whatever I/we add for https://github.com/google/iree/issues/6165 (`test.sh` for each sample that runs a sequence of commands without needing to fit natively into our CMake or Bazel configurations).

Reply:
Works with `iree_bytecode_module`, sent https://github.com/KoolJBlack/iree/pull/1